File: examples/11-leibniz.js
// https://niklas-heer.github.io/speed-comparison/
// https://github.com/niklas-heer/speed-comparison/blob/master/src/leibniz.js

let start_time = clock();
for (let loop = 1;; loop++) {
    let x = 1.0;
    let pi = 1.0;
    let rounds = 1000000;
    let i = 2;
    while (i < rounds + 2) {
        x *= -1;
        pi += x / (2 * i - 1);
        i++;
    }
    pi *= 4;
    gc();
    print((clock() - start_time) / loop, "secs, pi is", pi);
}

File: examples/11-leibniz.lua
require('os')
local start_time = os.clock()
local loop = 1
while true do
    local rounds = 1000000
    local x = 1
    local pi = 1.0
    local i = 2
    while i < rounds + 2 do
        x = -x
        pi = pi + (x / (2 * i - 1))
        i = i + 1
    end
    pi = pi * 4
    print((os.clock() - start_time) / loop, 'secs, pi is', pi)
    loop = loop + 1
end
